src/app/features/tech-record/create-batch-trl/components/batch-trl-details/batch-trl-details.component.ts

```diff
@@ -68,13 +71,23 @@ export class BatchTrlDetailsComponent implements OnDestroy {
 
   get vehicleForm(): FormGroup {
     return this.fb.group({
-      vin: [null, [Validators.minLength(3), Validators.maxLength(21)], [this.technicalRecordService.validateVin()]],
-      trailerId: ['', [Validators.minLength(7), Validators.maxLength(8), CustomValidators.alphanumeric()]]
+      vin: new CustomFormControl(
+        { name: 'vin', type: FormNodeTypes.CONTROL },
+        null,
+        [Validators.minLength(3), Validators.maxLength(21)],
+        this.technicalRecordService.validateForBatch()
+      ),
+      trailerId: ['', [Validators.minLength(7), Validators.maxLength(8), CustomValidators.alphanumeric()]],
+      systemNumber: ['']
     });
   }
 
-  getVin(group: AbstractControl): AbstractControl | null {
-    return group.get('vin');
+  validate(group: AbstractControl): void {
+    group.get('vin')!.updateValueAndValidity();
+  }
+
+  getVin(group: AbstractControl): CustomFormControl | null {
+    return group.get('vin') as CustomFormControl | null;
   }
```

Comment:
```suggestion
  get vinControl(): CustomFormControl | null {
    return this.vehicleForm.get('vin') as CustomFormControl | null;
  }
```

I know this `getVin` method was already here, however, I think it could be a getter instead of a method.

In the current state, it's slightly confusing since the `.ts` file of the component has the form, which gets passed to the `.html` file. The `html` file the calls this method with the formGroup in order to get the vin. The `.ts` file of the component already knows what the form group is and it's always static so I don't think there is a reason for the `.html` to pass the group in really.

Reply:
I don't think that suggestion would work because the current retrieves the vin control from the specific group it's tied to, whereas your suggestion would retrieve a new vin control that gets built in a new vehicleForm